File: src/main/java/be/vsol/vsol6/model/Organization.java
package be.vsol.vsol6.model;

import be.vsol.database.db;
import be.vsol.tools.json;
import be.vsol.vsol4.model.Vsol4Organization;

public class Organization extends Record {

    @json @db private String name, description;

    // Constructors

    public Organization() { }

    // TestConstructor
    public Organization(String name) {
        this.name = name;
    }

    public Organization(Vsol4Organization vsol4Organization) {
        super(vsol4Organization.getId(), vsol4Organization.getLastOpenedDate());
        name = vsol4Organization.getName();
        description = vsol4Organization.getDescription();
    }

    // Methods

    @Override public String toString() {
        return name;
    }

    // Getters

    public String getName() { return name; }

    public String getDescription() { return description; }

    // Setters

    public void setName(String name) { this.name = name;}

    public void setDescription(String description) { this.description = description;}

}

File: src/main/resources/js/model/Organization.ts
import {VsolRecord} from "./VsolRecord.js";

export class Organization extends VsolRecord {
    private readonly name : string;
    private readonly description : string;

    public constructor(id : string = null, name = "", description = "") {
        super(id);
        this.name = name;
        this.description = description;
    }

    public static from(src : any) {
        return src == null ? null : new Organization(src.id, src.name, src.description);
    }

    public toString() {
        return this.name;
    }

    // Getters

    public getName() { return this.name; }
    public getDescription() { return this.description; }

}